tiledb/sm/c_api/tiledb.cc

```diff
@@ -1437,29 +1437,16 @@ int32_t tiledb_array_get_open_timestamp_end(
 
 int32_t tiledb_array_delete(tiledb_ctx_t* ctx, const char* uri) {
   // Allocate an array object
-  tiledb_array_t* array = new (std::nothrow) tiledb_array_t;
-  try {
-    array->array_ = make_shared<tiledb::sm::Array>(
-        HERE(), ctx->resources(), tiledb::sm::URI(uri));
-  } catch (std::bad_alloc&) {
-    auto st = Status_Error(
-        "Failed to create TileDB array object; Memory allocation error");
-    delete array;
-    array = nullptr;
-    LOG_STATUS_NO_RETURN_VALUE(st);
-    save_error(ctx, st);
-    return TILEDB_OOM;
-  }
-
+  tiledb::sm::Array array(ctx->resources(), tiledb::sm::URI(uri));
```

Comment:
It would be better to continue to use `make_shared` here. We'll eventually want to do deletions through the facade object for `class Array` (which doesn't exist yet). For consistency, we'll need to allocate it like all the others.

As we rationalize array operations, we're going to want to move the deletion code out of this C API function and into the `sm/array` directory, where it can be made more consistent with everything else. At that point, we'll still open the array exclusively here, but then access the deletion function through the facade.